tools/render-test/render-d3d11.cpp

```diff
@@ -40,61 +40,124 @@ using namespace Slang;
 
 namespace renderer_test {
 
-struct D3DBinding
-{
-    ShaderInputType type;
-    InputBufferType bufferType; // Only valid if `type` is `Buffer`
-    ID3D11ShaderResourceView * srv = nullptr;
-    ID3D11UnorderedAccessView * uav = nullptr;
-    ID3D11Buffer * buffer = nullptr;
-    ID3D11SamplerState * samplerState = nullptr;
-    int binding = 0;
-    bool isOutput = false;
-    int bufferLength = 0;
-};
-struct D3DBindingState
+//static char const* vertexEntryPointName    = "vertexMain";
+//static char const* fragmentEntryPointName  = "fragmentMain";
+
+//static char const* vertexProfileName   = "vs_4_0";
+//static char const* fragmentProfileName = "ps_4_0";
+
+class D3D11Renderer : public Renderer, public ShaderCompiler
 {
-    List<D3DBinding> bindings;
-    int numRenderTargets = 0;
-};
+public:
+    // Renderer    implementation
+    virtual SlangResult initialize(void* inWindowHandle) override;
+    virtual void setClearColor(float const* color) override;
+    virtual void clearFrame() override;
+    virtual void presentFrame() override;
+    virtual SlangResult captureScreenShot(char const* outputPath) override;
+    virtual void serializeOutput(BindingState* state, const char * fileName) override;
+    virtual Buffer* createBuffer(BufferDesc const& desc) override;
+    virtual InputLayout* createInputLayout(InputElementDesc const* inputElements, UInt inputElementCount) override;
+    virtual BindingState * createBindingState(const ShaderInputLayout& layout) override;
+    virtual ShaderCompiler* getShaderCompiler() override;
+    virtual void* map(Buffer* buffer, MapFlavor flavor) override;
+    virtual void unmap(Buffer* buffer) override;
+    virtual void setInputLayout(InputLayout* inputLayout) override;
+    virtual void setPrimitiveTopology(PrimitiveTopology topology) override;
+    virtual void setBindingState(BindingState * state);
+    virtual void setVertexBuffers(UInt startSlot, UInt slotCount, Buffer* const* buffers, UInt const* strides, UInt const* offsets) override;    
+    virtual void setShaderProgram(ShaderProgram* inProgram) override;
+    virtual void setConstantBuffers(UInt startSlot, UInt slotCount, Buffer* const* buffers, UInt const* offsets) override;
+    virtual void draw(UInt vertexCount, UInt startVertex) override;
+    virtual void dispatchCompute(int x, int y, int z) override;
+    
+    // ShaderCompiler implementation
+    virtual ShaderProgram* compileProgram(ShaderCompileRequest const& request) override;
 
-//
+    protected:
 
+    struct BindingImpl
+    {
+        ShaderInputType type;
+        InputBufferType bufferType;                        // Only valid if `type` is `Buffer`
+        ID3D11ShaderResourceView * srv = nullptr;
```

Comment:
This change combines both the `Slang::ComPtr` introduction and the refactoring work on the renderers, so it is tough to see if there have been any functional changes made along the way (e.g., does the D3D11 renderer use `ComPtr` at all? It doesn't look like `BindingImpl` here does).

I'd advocate for splitting function and non-functional changes into distinct PRs as much as possible (e.g., "here's a PR with a bunch of typo fixes", "here's one with a bunch of renaming", "here's one with refactoring work (no functional changes)", and "here's the actual functional change I wanted to make all along (plus a test that verifies the change works as intended").

This approach is mostly to allow code reviews to be very focused when it matters (the functional change), and then to be very quick when it doesn't (I can glance at 1000 diffs that all change `Foo` to `FOO` and believe that it was done correctly).